### PR TITLE
Adjust thresholds for reasonable upgrade durations

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-1003](https://issues.redhat.com//browse/TRT-1003)

- The test is renamed to remove the value from the name, meaning we'll be able to do apples-to-apples comparisons in Sippy and Component Readiness.

- Thresholds are adjusted to the 30-day P99 values for 4.14. We've encounted some regressions, a bug was opened for s390x for example.

### CODE REVIEW

File: test/e2e/upgrade/upgrade.go [hunk 1]
```diff
@@ -359,10 +359,40 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
...
+	// OS, which would be the same as a minor upgrade, so we use the minor threshold for all values.
+	//
+	// [1] https://raw.githubusercontent.com/openshift/sippy/f546164c18db9a4a930cc64f43bcdcd35509e767/scripts/sql/test-duration-stats.sql
+	type limitLocator struct {
+		Platform configv1.PlatformType
+		Network  string
+	}
+	var upgradeDurationLimits = map[limitLocator]float64{
+		{configv1.AWSPlatformType, OVN}:       85,
+		{configv1.AWSPlatformType, SDN}:       130,
```
Comment (stbenjam) [inline test/e2e/upgrade/upgrade.go:386]: This value worries me, it's thrown off by just a small number of outliers. Here's the by-day stats for aws upgrade-minor, sdn:

```
 timestamp  |   percentile_50   |   percentile_99    |   percentile_100   
------------+-------------------+--------------------+--------------------
 2023-04-03 | 66.33668445480001 |  68.67009867101666 |  68.67009867101666
 2023-04-04 |    61.00122070645 |  65.33670738831667 |  65.33670738831667
 2023-04-05 | 65.66994640008333 |  66.83663089001666 |  66.83663089001666
 2023-04-06 |     71.1699914034 | 130.00100292186667 | 130.00100292186667
 2023-04-07 | 63.50128090998333 |  64.00166007301667 |  64.00166007301667
 2023-04-08 | 64.50178690341667 |  67.83420327996666 |  67.83420327996666
 2023-04-09 | 63.00180272506667 | 63.834119744966664 | 63.834119744966664
 2023-04-10 |    61.83355659865 |  62.50341595853333 |  62.50341595853333
 2023-04-11 | 62.50338116806667 |  63.66693479318334 |  63.66693479318334
 2023-04-12 | 62.83366054938333 |  66.83653658128334 |  66.83653658128334
 2023-04-13 | 65.50332776381667 | 123.83383404793334 | 123.83383404793334
 2023-04-14 | 62.33785857713334 |      63.0011904242 |      63.0011904242
 2023-04-15 | 62.16696633663333 |  65.17051039851667 |  65.17051039851667
 2023-04-16 | 62.50331522281667 |  65.33356643743333 |  65.33356643743333
 2023-04-17 | 61.83411498416667 |  64.16791090768334 |  64.16791090768334
 2023-04-18 | 60.66809686411667 |  63.83653952776667 |  63.83653952776667
 2023-04-19 | 63.50436424363333 |  67.67001490021667 |  67.67001490021667
 2023-04-20 | 67.83669018163333 |  70.16891166096666 |  70.16891166096666
 2023-04-21 | 68.83359635573333 |  71.33662739818332 |  71.33662739818332
 2023-04-22 |    68.00084324615 |  72.83777085256666 |  72.83777085256666
 2023-04-23 |     66.8351065497 |  71.33723795998334 |  71.33723795998334
 2023-04-24 |     68.8345876203 |  70.00339408296666 |  70.00339408296666
 2023-04-25 |    68.67053672035 |     71.67106336005 |     71.67106336005
 2023-04-26 | 66.66754525656667 |  72.83672768596666 |  72.83672768596666
 2023-04-27 | 67.16839730008333 | 133.50677713456668 | 133.50677713456668
 2023-04-28 | 69.66994769221667 |  71.33660977888333 |  71.33660977888333
 2023-04-29 | 68.83508444216666 |  70.00376385868334 |  70.00376385868334
 2023-04-30 | 67.50338064378333 |  71.33662319876666 |  71.33662319876666
 2023-05-01 | 69.67003445261666 |  74.16994683001667 |  74.16994683001667
 2023-05-02 | 69.16996374988334 |  70.00433161243333 |  70.00433161243333
 2023-05-03 | 65.00124213528333 |  67.66696476931666 |  67.66696476931666
```

There's a handful of days (not clustered together) where the p99 is >120.  If we throw them away, we get a value around 75 which seems to make more sense to me.

**Comment by deads2k** [comment]
looks reasonable to me.  I'd like to see a meeting with OTA and @sdodson establishing OTA as accountable for upgrade duration.

leaving lgtm with
/assign @dgoodwin

**Comment by stbenjam** [comment]
I want to make sure the adjusted thresholds aren't going to botch payloads.

/payload 4.14 nightly blocking

File: test/e2e/upgrade/upgrade.go [hunk 2]
```diff
@@ -569,10 +596,10 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
 			// record whether the cluster was fast or slow upgrading.  Don't fail the test, we still want signal on the actual tests themselves.
 			upgradeEnded := time.Now()
 			upgradeDuration := upgradeEnded.Sub(upgradeStarted)
-			testCaseName := fmt.Sprintf("[sig-cluster-lifecycle] cluster upgrade should complete in %0.2f minutes", durationToSoftFailure.Minutes())
+			testCaseName := fmt.Sprintf("[sig-cluster-lifecycle] cluster upgrade should complete in a reasonable time")
 			failure := ""
-			if upgradeDuration > durationToSoftFailure {
-				failure = fmt.Sprintf("%s to %s took too long: %0.2f minutes", action, versionString(desired), upgradeDuration.Minutes())
+			if upgradeDuration > upgradeDurationLimit {
+				failure = fmt.Sprintf("%s to %s took too long: %0.2f minutes (for this platform/network, it should be less than %0.2f minutes)", action, versionString(desired), upgradeDuration.Minutes(), upgradeDurationLimit.Minutes())
```
Comment (dgoodwin) [inline test/e2e/upgrade/upgrade.go:602]: Just a side note you could consider, if the test duration you already looked at isn't sufficient, extracting test metadata here on import and then a grafana panel would be a pretty quick path to charting upgrade times by day.

Comment (stbenjam) [reply test/e2e/upgrade/upgrade.go:602]: We have the by-day view in Sippy https://sippy.dptools.openshift.org/sippy-ng/tests/4.14/analysis?test=Cluster%20upgrade.%5Bsig-cluster-lifecycle%5D%20Cluster%20completes%20upgrade

If we wire up grafana to postgres we should be able to see it without any extra data collection

**Comment by stbenjam** [comment]
/hold cancel

Payload jobs look fine-ish (not failures of this test anyway)

**Review by wking — APPROVED** [review]
/approve

My main concern with this kind of thing is how we ratchet things back down when updates start taking "too long", but [OCPBUGS-13059](https://issues.redhat.com/browse/OCPBUGS-13059)  will give us a fresh take on that experience.

The required [e2e-gcp-ovn][1] is failing on:

```
DaemonSet/openshift-e2e-loki/loki-promtail/container/prod-bearer-token does not have a ... request
```

Possibly openshift/release#38914 needs some tweaking?

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27905/pull-ci-openshift-origin-master-e2e-gcp-ovn/1654189722072780800

**Pull request merged**